packages/utils/src/crypto/ec-utils.ts

```diff
@@ -121,13 +136,18 @@ async function decrypt(privateKey: string, encrypted: string): Promise<string> {
   try {
     return await EthCrypto.decryptWithPrivateKey(privateKey, EthCrypto.cipher.parse(encrypted));
   } catch (e) {
-    if (e.message === 'Bad private key') {
+    if (
+      e.message === 'Bad private key' ||
+      e.message === 'Expected private key to be an Uint8Array with length 32'
+    ) {
       throw new Error('The private key must be a string representing 32 bytes');
     }
     if (
       e.message === 'public key length is invalid' ||
+      e.message === 'Expected public key to be an Uint8Array with length [33, 65]' ||
       e.message === 'Bad MAC' ||
-      e.message === 'the public key could not be parsed or is invalid'
+      e.message === 'the public key could not be parsed or is invalid' ||
+      e.message === 'Public Key could not be parsed'
     ) {
```

Comment:
the error messages have changed. I added the new ones of top of the old ones  just in case.